src/features/hotspots/root/HotspotsScreen.tsx

```diff
@@ -46,12 +87,8 @@ const HotspotsScreen = () => {
   })
 
   useEffect(() => {
-    // dispatch(fetchHotspotsData()) will trigger an update to hotspots
-    // anytime hotspots update, update rewards and following
-    // Separating these calls allows the UI to respond more quickly
-    dispatch(fetchFollowedHotspotsFromBlock())
-    dispatch(fetchRewards())
-  }, [hotspots, dispatch])
+    dispatch(fetchRewards({ fetchType: fleetModeEnabled ? 'followed' : 'all' }))
```

Comment:
is there some state they could get into where they follow too many hotspots and it bogs down too much to where they can't get in to unfollow them? Should there be an upper limit on followed hotspots? at least until we have the API sorted out?

Reply:
Yes, we discussed this at one point and decided not to handle it for the moment. I will raise the question again and can address it in a separate PR if needed.